arch/x86/kernel/slaunch.c

```diff
@@ -466,6 +466,57 @@ void __init slaunch_setup(void)
 		slaunch_setup_intel();
 }
 
+#define MSG_BUFFER_LEN 20
+#define MSG_ERROR "Mapping Error\n"
+
+#define DECLARE_PUB_READ(reg_name, reg_offset, reg_size)					\
+static ssize_t reg_name##_read(struct file *flip, char *buffer, size_t len, loff_t *offset) {	\
+	char msg_buffer[MSG_BUFFER_LEN];							\
+	txt_info_to_buffer(reg_offset, reg_size, msg_buffer);					\
+	printk(KERN_INFO "%s", msg_buffer);							\
```

Comment:
checkpatch is going to complain about using printk. Use something like pr_info instead.

Reply:
fixed - just removed the print statement entirely